### PR TITLE
fix(tablekit-react): fix table row styles

### PR DESCRIPTION
**Pull request by SashaShostyr**
- Remove transition-in animation;
- Add cursor pointer on hover.

Relates to MOPS-1021, MOPS-1022.

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @tablecheck/tablekit-core@2.0.1-canary.183.5021982221.0
  npm install @tablecheck/tablekit-css@2.0.1-canary.183.5021982221.0
  npm install @tablecheck/tablekit-react-css@2.0.1-canary.183.5021982221.0
  npm install @tablecheck/tablekit-react-datepicker@2.0.1-canary.183.5021982221.0
  npm install @tablecheck/tablekit-react-fit-content-textarea@2.0.1-canary.183.5021982221.0
  npm install @tablecheck/tablekit-react-select@2.0.1-canary.183.5021982221.0
  npm install @tablecheck/tablekit-react@2.0.1-canary.183.5021982221.0
  # or 
  yarn add @tablecheck/tablekit-core@2.0.1-canary.183.5021982221.0
  yarn add @tablecheck/tablekit-css@2.0.1-canary.183.5021982221.0
  yarn add @tablecheck/tablekit-react-css@2.0.1-canary.183.5021982221.0
  yarn add @tablecheck/tablekit-react-datepicker@2.0.1-canary.183.5021982221.0
  yarn add @tablecheck/tablekit-react-fit-content-textarea@2.0.1-canary.183.5021982221.0
  yarn add @tablecheck/tablekit-react-select@2.0.1-canary.183.5021982221.0
  yarn add @tablecheck/tablekit-react@2.0.1-canary.183.5021982221.0
  ```
</details>

### CODE REVIEW

File: system/core/src/components/Table.ts [hunk 1]
```diff
@@ -23,6 +23,8 @@ export const baseStyles = css`
 
     & tr[data-pseudo=''],
     & tr:hover {
+      cursor: pointer;
```
Comment (SimeonC) [inline system/core/src/components/Table.ts:26]: By default, table rows should have the default cursor. Not all table rows are clickable.

File: system/core/src/components/Table.ts [hunk 2]
```diff
@@ -23,6 +23,8 @@ export const baseStyles = css`
...
+      transition: none;
```
Comment (SimeonC) [inline system/core/src/components/Table.ts:27]: Rather than setting the transition to none here - just remove the transition in `& tr`

**Review by SimeonC — CHANGES_REQUESTED** [review]

File: system/core/src/components/Table.ts (hunk 2 above)
Comment (SashaShostyr) [reply system/core/src/components/Table.ts:27]: I guess the request is to remove only transition-in animation not all. https://tablecheck.slack.com/archives/CCXQ4RAUE/p1682681708436729

File: system/core/src/components/Table.ts (hunk 1 above)
Comment (SashaShostyr) [reply system/core/src/components/Table.ts:26]: Updated

**Review by SimeonC — APPROVED** [review]

**Review by nkgardose — APPROVED** [review]

**Pull request merged**